ambry-replication/src/main/java/com/github/ambry/replication/RemoteReplicaInfo.java

```diff
@@ -62,6 +63,9 @@
   private long reEnableReplicationTime = 0;
   private ReplicaThread replicaThread;
 
+  // Metadata response information received for this replica in the most recent replication cycle.
+  private ReplicaThread.ExchangeMetadataResponse exchangeMetadataResponse;
```

Comment:
Add a little more comments here regarding what we need from `ExchangeMetadataResponse` for leader-based replication.